h2/stream.py

```diff
@@ -787,15 +789,14 @@ def push_stream_in_band(self, related_stream_id, headers, encoder):
         # Because encoding headers makes an irreversible change to the header
         # compression context, we make the state transition *first*.
 
-        # This does not trigger any events.
```

Comment:
That may mean we need a third event here, which you can use to help calculate your state tuple.